### PR TITLE
[Course Task] Doc devicetwin

### PR DESCRIPTION
**Pull request by jxs1211**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
docs update

* **What is the current behavior?** (You can also link to an open issue here)
none

* **What is the new behavior (if this is a feature change)?**
[123](https://github.com/kubeedge/community/issues/123)

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
none

* **Other information**:
none

### CODE REVIEW

**Comment by wackxu** [comment]
please rebase your branch with master

**Comment by jxs1211** [comment]
> please rebase your branch with master
> 
> 

updated

File: content/en/docs/architecture/edge/devicetwin.md [hunk 1]
```diff
@@ -79,43 +79,33 @@ The major functions performed by this module are:-
...
 
-**dealMembershipDetail**:   dealMembershipDetail() provides the membership details of the edge node, providing information
-                            about the devices associated with the edge node, after removing the membership details of
-                            recently removed devices.
-- The eventbus module receives the message that arrives on the subscribed topic,the message is then forwarded  to the
-devicetwin controller which further forwards it to the membership module.
-- The membership  module adds devices that are mentioned in the message, removes
-devices that that are not present in the cache.
-- After updating the details of the devices a  message is sent to the communication module of the device twin.
+**dealMembershipDetail**:   dealMembershipDetail() provides the membership details of the edge node, providing information about the devices associated with the edge node, after removing the membership details of                        recently removed devices.
```
Comment (kevin-wangzefeng) [inline content/en/docs/architecture/edge/devicetwin.md:105]:
```suggestion
**dealMembershipDetail**:   dealMembershipDetail() provides the membership details of the edge node, providing information about the devices associated with the edge node, after removing the membership details of recently removed devices.
```

File: content/en/docs/architecture/edge/devicetwin.md [hunk 2]
```diff
@@ -207,48 +197,48 @@ The following are the action callbacks which can be performed by the communicati
...
+   - The edgehub component sends the message to the device twin controller which forwards the message to the device module.
+   - The device module updates the device attribute details into the database after which, the device module sends the result of the device attribute update to be published
+   to the eventbus component through the communicate module of devicetwin. The eventbus component further publishes the result on the specified topic.
 
   ![Device Update](/img/devicetwin/device-update.png)
 
  **dealDeviceStateUpdate**:  dealDeviceStateUpdate() deals with the operations to be performed when a device status update is encountered.
                              It updates the state of the device as well as the last online time of the device in the database.
                              It also sends the update state result, through the communication module,  to the cloud through the edgehub module and to the  eventbus module which in turn
                              publishes the result on the specified topic of the MQTT broker.
```
Comment (kevin-wangzefeng) [inline content/en/docs/architecture/edge/devicetwin.md:225]: Could you also fix the indent here?

File: content/en/docs/architecture/edge/devicetwin.md [hunk 3]
```diff
@@ -79,43 +79,33 @@ The major functions performed by this module are:-
...
    - dealMembershipDetail
 
-**dealMembershipGet**:    dealMembershipGet()  gets the information  about the devices associated with the particular edge node, from the cache.
-- The eventbus first receives a message on its subscribed topic (membership-get topic).
-- This message arrives at the  devicetwin controller, which further sends the message to membership module.
-- The membership module gets the devices associated with the edge node from the cache (context) and sends the information to the communication module.
- It also handles errors that may arise while performing the  aforementioned process and sends the error to the communication module instead of device details.
-- The communication module sends the information to the  eventbus component which further publishes the result on the
- specified MQTT topic (get membership result topic).
+**dealMembershipGet**:    dealMembershipGet()  gets the information  about the devices associated with the particular edge node from the cache.
```
Comment (kevin-wangzefeng) [inline content/en/docs/architecture/edge/devicetwin.md:88]:
```suggestion
**dealMembershipGet**:  dealMembershipGet() gets the information  about the devices associated with the particular edge node from the cache.
```

File: content/en/docs/architecture/edge/devicetwin.md [hunk 4]
```diff
@@ -79,43 +79,33 @@ The major functions performed by this module are:-
...
-                            It adds the devices, that were newly added, to the edge group and removes the devices, that were removed,
-                            from the edge group and updates device details, if they have been altered or updated.
-- The edgehub module receives the membership update message from the cloud and forwards the message
-to devicetwin controller which further forwards it to the membership module.
-- The membership  module adds devices that are newly added, removes devices that have been recently
-deleted and also updates the devices that were already existing in the database as well as in the cache.
-- After updating the details of the devices a  message is sent to the communication module of the device twin, which sends the message to eventbus module to be published on the given MQTT topic.
+**dealMembershipUpdated**:  dealMembershipUpdated() updates the membership details of the node. It adds the devices, that were newly added, to the edge group and removes the devices, that were removed, from the edge group and updates device details, if they have been altered or updated.
```
Comment (kevin-wangzefeng) [inline content/en/docs/architecture/edge/devicetwin.md:97]:
```suggestion
**dealMembershipUpdated**: dealMembershipUpdated() updates the membership details of the node. It adds the devices, that were newly added, to the edge group and removes the devices, that were removed, from the edge group and updates device details, if they have been altered or updated.
```

**Review by kevin-wangzefeng — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**